### DISCUSSION TITLE
Editing a reference to a doc, live. (aka Transclusion)

### DISCUSSION THREAD

**Discussion by quesada** [zim-desktop-wiki/zim-desktop-wiki#1907] [Ideas]
The fancy new generation of 'graph notetakers' loves a term: Transclusion. 

https://en.wikipedia.org/wiki/Transclusion

Best way to understand it is to see it at work:
![image](https://user-images.githubusercontent.com/182207/152638110-03a3c9a6-14fc-4dba-91da-95e01ba6299b.png)
(if the animation doesn't work, go to https://nesslabs.com/remnote-featured-tool and scroll mid page.

Use cases:
1. Writing multiple books with many chapters where chapters are collections of notes. You update a note while writing one book, would love if the changes would automatically propagate to another book that uses that note. 
Zim doesn't even have a way to concatenate docs into a 'master' document, let alone editing those sections live from the master. So  this use case is a bit of a stretch
2. 'backlinks' showing context (text where this concept is used) could be edited live, in the same place where they are showing the text. Workflowy can do that, obsidian cannot. Zim doesn't show any context in the backlink panel so this is also a stretch (right now)

Having an 'empty' note in zim happens often for me. That is, I have an entry in the index that is only a title, with no body text. The sidebar shows those as greyed out notes. This is what other tools call 'linking a paragraph'. And these show up when linking (ctrl L).

Everything is a 'rem' in remnote (that is, every paragraph on every note ever). This is a very different design from what zim uses and I suspect a precursor for this idea of transclusion.

The Problem to implement transclusion is that this requires a 'serious' database where every paragraph is indexed, not just the note files' title. This is what obsidian is doing. My guess, of course. And this might not be the route @jaap-karssenberg has in mind, as he has supremely good taste in software architecture and likes simplicity.

This is just me thinking outloud :) Sorry about the rant.

**Comment by introt** [comment]
Re: transclusion: #1818
